nix/haskell/mk-hackage.nix

```diff
@@ -0,0 +1,132 @@
+{ gzip
+, runCommand
+, lib
+, nix-tools
+}:
+let
+  mkPackageSpec = src:
+    with lib;
+    let
+      cabalFiles = concatLists (mapAttrsToList
+        (name: type: if type == "regular" && hasSuffix ".cabal" name then [ name ] else [ ])
+        (builtins.readDir src));
+
+      cabalPath =
+        if length cabalFiles == 1
+        then src + "/${builtins.head cabalFiles}"
+        else builtins.abort "Could not find unique file with .cabal suffix in source: ${src}";
+      cabalFile = builtins.readFile cabalPath;
+      parse = field:
+        let
+          lines = filter (s: builtins.match "^${field} *:.*$" (toLower s) != null) (splitString "\n" cabalFile);
+          line =
+            if lines != [ ]
+            then head lines
+            else builtins.abort "Could not find line with prefix ''${field}:' in ${cabalPath}";
+        in
+        replaceStrings [ " " ] [ "" ] (head (tail (splitString ":" line)));
+      pname = parse "name";
+      version = parse "version";
+    in
+    { inherit src pname version; };
+
+  mkHackageDir = { pname, version, src }:
```

Comment:
Where is this from? Please put the link so that we know where to look when something breaks

Reply:
I'm not sure honestly, it's something that we've been copy-pasting around for years at this point. Probably https://github.com/mlabs-haskell/mlabs-tooling.nix/blob/main/mk-hackage.nix is a good reference point but it's not maintained